src/game_inventory.cpp

```diff
@@ -134,18 +134,112 @@ int game::inv_for_id( const itype_id &id, const std::string &title )
     }, string_format( _( "You don't have a %s." ), item::nname( id ).c_str() ) );
 }
 
-int game_menus::inv::take_off( player &p )
+class armor_inventory_preset: public inventory_selector_preset
 {
-    return g->inv_for_filter( _( "Take off item" ), [ &p ]( const item & it ) {
-        return p.is_worn( it );
-    }, _( "You don't wear anything." ) );
+    public:
+        armor_inventory_preset( const std::string &color ) {
+
+            static const auto get_number_string = [ &color ]( int number ) {
```

Comment:
`color` is bound to a temporary object via the constructor call in `game_menus::inv::wear`. This *static* lambda is create once and when `wear` is called again, it's still there and has still a *reference* the previously created string object, which is long gone.

Seriously: what's so bad about copying a simple string, especially as it's only a short color name?

And by the way: this lambda should not be static at all. It's capturing the color, but the color can be different on *each* call to this constructor, it's a parameter after all.